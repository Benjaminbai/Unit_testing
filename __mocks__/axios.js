
const mockAxios = jest.requireActual('Axios')

export default mockAxios;