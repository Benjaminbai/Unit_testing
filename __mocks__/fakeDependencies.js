jest.mock('react-router-dom', () => {
    const originalModule = jest.requireActual('react-router-dom');

    return {
        __esModule: true,
        ...originalModule,
        useLocation: jest.fn().mockReturnValue({
            pathname: '',
            search: '',
            hash: '',
            state: null,
            key: '',
        }),
        useHistory: () => ({
            push: jest.fn(),
            location: {}
        }),
        Redirect: jest.fn().mockReturnValue({}),
    }
})
window.matchMedia = jest.fn().mockImplementation((query) => {
    const obj = {
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(),
        removeListener: jest.fn(),
    };

    return obj;
})

jest.mock('react', () => ({
    ...jest.requireActual('react'),
    useLayoutEffect: jest.requireActual('react').useEffect
}))