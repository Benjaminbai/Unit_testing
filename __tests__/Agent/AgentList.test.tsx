import '../../__mocks__/fakeDependencies'
import React from 'react'
import { mount } from 'enzyme'
import AgentList from '../../pages/Agent/AgentList'
import { BrowserRouter as Router } from 'react-router-dom'
import { render as RTLRender, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import API from '../../api/Agent'

const AgentListComponent = mount(<Router><AgentList /></Router>)

describe('<AgentList />', () => {
    test('expect the AgentList exist', () => {
        expect(AgentListComponent.exists()).toBe(true)
    })

    test('expect AgentList unmount', () => {
        AgentListComponent.unmount()
        expect(() => {
            AgentListComponent.instance()
        }).toThrow()
    })

    // 快照
    test('the source code changed or not', () => {
        expect(AgentListComponent).toMatchSnapshot()
    })

    test('api listRegistryAgent', () => {
        const spyFn = jest.spyOn(API, 'listRegistryAgent')
        RTLRender(<Router><AgentList /></Router>)
        userEvent.click(screen.getByText(/搜 索/i))
        expect(spyFn).toBeCalled()
    })
})

