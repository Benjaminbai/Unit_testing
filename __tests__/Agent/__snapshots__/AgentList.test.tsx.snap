// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AgentList /> the source code changed or not 1`] = `null`;
